### PR TITLE
Chameleon PDA's

### PR DESCRIPTION
**Pull request by Rainfey**
## About the PR
Added a chameleon PDA to the chameleon kit
Make chameleon clothing use Slot and EntityWhitelist to select valid variants, rather than Slot and  the WhitelistChameleon tag

## Why / Balance
Major equippable item that was missing from the chameleon kit
Now allows for chameleon clothing items to support subsets of items that fit in their slot (ie you could have Chameleon Magboots which only support magboot variants)

## Technical details
Required adding a handler for PDA's snowflake method of storing their appearance (field on the PDAComponent)
Moved some functionality into SharedChameleonClothing so the server can check if a target is valid
ChameleonClothingComponent and ChameleonBoundUserInterfaceState now contain an EntityWhitelist
Added support for the recent IdentityBlocker partial coverage PR #24741 

## Media
Screenshot 2024-01-30 165959

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Chameleon PDA

### CODE REVIEW

**Comment by potato1234x** [comment]
the agent id chameleon should probably be fixed instead of removed, knowing the maintainers they probably wont merge until fixed

**Comment by Rainfey** [comment]
Well i wanted to avoid it - but i would also prefer having both chameleon ID and PDA.
Drafting until i rewrite things to support that.
Probably going to add an EntityWhitelist to each chameleon item.

File: Content.Client/Clothing/Systems/ChameleonClothingSystem.cs
Comment (metalgearsloth) [inline Content.Client/Clothing/Systems/ChameleonClothingSystem.cs:39]: TryIndex

File: Content.Shared/Clothing/Components/ChameleonClothingComponent.cs
Comment (metalgearsloth) [inline Content.Shared/Clothing/Components/ChameleonClothingComponent.cs:49]: You can just network it in the component and not need it in the BUI state at all.

File: Content.Shared/Clothing/EntitySystems/SharedChameleonClothingSystem.cs
Comment (metalgearsloth) [inline Content.Shared/Clothing/EntitySystems/SharedChameleonClothingSystem.cs:73]: TryIndex

File: Content.Shared/PDA/SharedPdaSystem.cs
Comment (metalgearsloth) [inline Content.Shared/PDA/SharedPdaSystem.cs:69]: This looks unused and it's also undocumented.

File: Content.Shared/Clothing/EntitySystems/SharedChameleonClothingSystem.cs
Comment (metalgearsloth) [inline Content.Shared/Clothing/EntitySystems/SharedChameleonClothingSystem.cs:190]: Why not valuelist if you're just going to return a slow ienumerable anyway?

File: Content.Shared/Whitelist/EntityWhitelist.cs
Comment (metalgearsloth) [inline Content.Shared/Whitelist/EntityWhitelist.cs:107]: You never use the factory.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Pull request closed without merging**
